### DISCUSSION TITLE
Select Component: Selected value doesn't show up

### DISCUSSION THREAD

**Discussion by jcsco** [mantinedev/mantine#3911] [Q&A] [answered]
Thaks for providing this great framework. I'm trying to use it for a prototype at work. I'm using the Select component on two pages and both are mostly identical with label names being different:

This is how I'm building the selectData:
```
        Object.entries(continuationEventData).forEach(([key, value]) => {
            selectData.push({
                "value": value,
                "label": key.replaceAll("+", ".").replaceAll("_", ""),
                "key": key,
                "id": key
            })
        });

```

```             

 <Select
                    style={{ minWidth: 500 }}
                    label={"Transition key"}
                    searchable
                    nothingFound="No options"
                    limit={50}
                    required data={selectData}
                    value={continuationDiscoveryValue}
                    onChange={setContinuationDiscoveryValue}
/>

```

On the first page, when I select the label of the dropdown, it does show what is selected. When I navigate to the second page, I see the problem where after selecting a label the box shows blank. In the background though, things are working as expected and the value does indeed get stored in the `continuationDiscoveryValue` variable. 
<img width="699" alt="image" src="https://user-images.githubusercontent.com/84595813/227807938-4c5f1103-ce50-4b8e-a4b7-cd63f4b92c65.png">

Here is the video of the problem: https://filebin.net/843bel7cudq3d3qy

Thanks for your help

**Comment by rtivital** [comment]
Select value must be in the `data`. For example:

```tsx
<Select 
  value="react"
  data={[{ value: 'react', label: 'React' }]}
/>
```

If an item cannot be found in the data array, then it will not be displayed.

**Reply by jcsco** [reply]
@rtivital  Thanks for your response. I do have it in the data 
```
{"value":"e114c8a2-e805-441f-8fbc-ad042d5c73a2","label":"source.segments.0.orderId","key":"_source+segments+0+orderId","id":"_source+segments+0+orderId"}
```
I'm using it exactly as per the documentation here: https://mantine.dev/core/select/#controlled .

**Comment by jcsco** [comment, marked as answer]
This is something about how I'm using it.  The following code block appears to be the problem. I'm using `useEffect` to wait for data to be rendered and then I build the `selectData` to build label and value data. I thought this would create the correct dependency. But just moving it out of the use effect seems to fix the issue. I'm a react newbie so can't exactly understand whats the problem here.

```
    useEffect(() => {
        //console.log("Event Data =>", JSON.stringify(continuationEventData))
        Object.entries(continuationEventData).forEach(([key, value]) => {
            selectData.push({
                "value": value,
                "label": key.replaceAll("+", ".").replaceAll("_", ""),
                "key": key
            })
        });
    }, [continuationEventData])
```
